src/frequenz/client/dispatch/_client.py

```diff
@@ -255,3 +267,31 @@ async def delete(self, dispatch_id: int) -> None:
         """
         request = DispatchDeleteRequest(id=dispatch_id)
         await self._stub.DeleteMicrogridDispatch(request)  # type: ignore
+
+    async def _try_fetch_created_dispatch(
+        self,
+        request: DispatchCreateRequest,
+    ) -> Dispatch | None:
+        """Try to fetch the created dispatch.
+
+        Will return the created dispatch if it was found, otherwise None.
+
+        Args:
+            request: The dispatch create request.
+
+        Returns:
+            Dispatch: The created dispatch, if it was found.
+        """
+        async for dispatch in self.list(microgrid_id=request.microgrid_id):
+            found = True
+            for key, value in request.__dict__.items():
+                if key == "start_time":
+                    value = rounded_start_time(value)
+
+                if not (found := getattr(dispatch, key) == value):
```

Comment:
Since python 3.11 `getattr` can be much slower because they get in the way of the adaptive interpreter and other optimizations.  It is too bad `api.CreateDispatch` doesn't return a dispatch ID, or the entire dispatch.

Won't be a bad idea to implement this in the api level, I think, so that you can avoid additional calls to list and then filter.

Reply:
> Won't be a bad idea to implement this in the api level, I think, so that you can avoid additional calls to list and then filter.

That's all in the works, this is just to provide the feature before the API is ready. That way we can later add it and clients don't need to change.